File: Maps_Project.Web/Modules/ServerTypes/Trip/BusinessTripColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { BusinessTripRow } from "./BusinessTripRow";

export interface BusinessTripColumns {
    Id: Column<BusinessTripRow>;
    Title: Column<BusinessTripRow>;
    Description: Column<BusinessTripRow>;
}

export class BusinessTripColumns extends ColumnsBase<BusinessTripRow> {
    static readonly columnsKey = 'Trip.BusinessTrip';
    static readonly Fields = fieldsProxy<BusinessTripColumns>();
}

File: Maps_Project.Web/Modules/Trip/BusinessTrip/BusinessTripColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace Maps_Project.Trip.Columns;

[ColumnsScript("Trip.BusinessTrip")]
[BasedOnRow(typeof(BusinessTripRow), CheckNames = true)]
public class BusinessTripColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Title { get; set; }
    public string Description { get; set; }
}